Crash when refreshing AWS state - nil security group rule?

Using version 0.6, can't tell from the crash which particular rule it is in our config (we have a lot).
```
* 1 error(s) occurred:
panic: runtime error: invalid memory address or nil pointer dereference
2015/07/25 14:32:15 terraform-provider-aws: [signal 0xb code=0x1 addr=0x0 pc=0x102529]
2015/07/25 14:32:15 terraform-provider-aws: 
2015/07/25 14:32:15 terraform-provider-aws: goroutine 270 [running]:
2015/07/25 14:32:15 terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsSecurityGroupRuleRead(0xc2081960c0, 0x6a5700, 0xc20806a2d0, 0x0, 0x0)
2015/07/25 14:32:15 terraform-provider-aws: 	/private/tmp/terraform20150702-45875-1xz55ha/terraform-0.6.0/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_security_group_rule.go:164 +0xa29
2015/07/25 14:32:15 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc208194180, 0xc208120f60, 0x6a5700, 0xc20806a2d0, 0xc208049040, 0x0, 0x0)
2015/07/25 14:32:15 terraform-provider-aws: 	/private/tmp/terraform20150702-45875-1xz55ha/terraform-0.6.0/src/github.com/hashicorp/terraform/helper/schema/resource.go:209 +0x475
2015/07/25 14:32:15 terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc208166810, 0xc208184640, 0xc208120f60, 0x6a6b40, 0x0, 0x0)
2015/07/25 14:32:15 terraform-provider-aws: 	/private/tmp/terraform20150702-45875-1xz55ha/terraform-0.6.0/src/github.com/hashicorp/terraform/helper/schema/provider.go:187 +0x1c7
2015/07/25 14:32:15 terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Refresh(0xc2081725a0, 0xc20818a560, 0xc20818a7d0, 0x0, 0x0)
2015/07/25 14:32:15 terraform-provider-aws: 	/private/tmp/terraform20150702-45875-1xz55ha/terraform-0.6.0/src/github.com/hashicorp/terraform/rpc/resource_provider.go:345 +0x73
2015/07/25 14:32:15 terraform-provider-aws: reflect.Value.call(0x858000, 0xa37120, 0x13, 0xaa2bc0, 0x4, 0xc2081cbf28, 0x3, 0x3, 0x0, 0x0, ...)
2015/07/25 14:32:15 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/reflect/value.go:419 +0x10e5
2015/07/25 14:32:15 terraform-provider-aws: reflect.Value.Call(0x858000, 0xa37120, 0x13, 0xc2081cbf28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/07/25 14:32:15 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/reflect/value.go:296 +0xbc
2015/07/25 14:32:15 terraform-provider-aws: net/rpc.(*service).call(0xc208194740, 0xc2081946c0, 0xc208128378, 0xc2080b3100, 0xc2081a6d60, 0x6a6ae0, 0xc20818a560, 0x16, 0x6a6b40, 0xc20818a7d0, ...)
2015/07/25 14:32:15 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:382 +0x1f7
2015/07/25 14:32:15 terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
2015/07/25 14:32:15 terraform-provider-aws: 	/usr/local/Cellar/go/1.4.2/libexec/src/net/rpc/server.go:476 +0x44a
```

--- a/builtin/providers/aws/resource_aws_security_group_rule.go
+++ b/builtin/providers/aws/resource_aws_security_group_rule.go
@@ -152,7 +152,9 @@ func resourceAwsSecurityGroupRuleRead(d *schema.ResourceData, meta interface{}) 
 	sg_id := d.Get("security_group_id").(string)
 	sg, err := findResourceSecurityGroup(conn, sg_id)
 	if err != nil {
+		log.Printf("[DEBUG] Error finding Secuirty Group (%s) for Rule (%s): %s", sg_id, d.Id(), err)
 		d.SetId("")
+		return nil
 	}
 
 	var rule *ec2.IPPermission
@@ -256,6 +258,7 @@ func findResourceSecurityGroup(conn *ec2.EC2, id string) (*ec2.SecurityGroup, er
 	if err != nil {
 		return nil, err
 	}
+
 	if resp == nil || len(resp.SecurityGroups) != 1 || resp.SecurityGroups[0] == nil {
 		return nil, fmt.Errorf(
 			"Expected to find one security group with ID %q, got: %#v",

--- a/builtin/providers/aws/resource_aws_security_group_rule_test.go
+++ b/builtin/providers/aws/resource_aws_security_group_rule_test.go
@@ -105,7 +105,7 @@ func TestIpPermissionIDHash(t *testing.T) {
 	}
 }
 
-func TestAccAWSSecurityGroupRule_Ingress(t *testing.T) {
+func TestAccAWSSecurityGroupRule_Ingress_VPC(t *testing.T) {
 	var group ec2.SecurityGroup
 
 	testRuleCount := func(*terraform.State) error {
@@ -142,7 +142,7 @@ func TestAccAWSSecurityGroupRule_Ingress(t *testing.T) {
 	})
 }
 
-func TestAccAWSSecurityGroupRule_IngressClassic(t *testing.T) {
+func TestAccAWSSecurityGroupRule_Ingress_Classic(t *testing.T) {
 	var group ec2.SecurityGroup
 
 	testRuleCount := func(*terraform.State) error {

